forge/db/models/ProjectSnapshot.js

```diff
@@ -51,15 +51,31 @@ module.exports = {
         const self = this
         return {
             static: {
-                byId: async function (id) {
-                    // This returns the full snapshot - including settings and flows
-                    // This should _only_ be used when getting all of that information
+                byId: async function (id, { includeFlows = true, includeSettings = true } = {}) {
+                    // By default, this returns the full snapshot - including settings and flows
+                    // This should _only_ be used when getting all of that information.
+                    // Otherwise use the options to disable retrieving flows/settings
                     if (typeof id === 'string') {
                         id = M.ProjectSnapshot.decodeHashid(id)
                     }
-                    return self.findOne({
-                        where: { id }
+                    const toExclude = []
+                    if (!includeFlows) {
+                        toExclude.push('flows')
+                    }
+                    if (!includeSettings) {
+                        toExclude.push('settings')
+                    }
+                    let attributes = null
+                    if (toExclude.length > 0) {
+                        attributes = {
+                            exclude: toExclude
+                        }
+                    }
+                    const r = await self.findOne({
```

Comment:
This can return directly, no need for r

Reply:
I added it in [85bf808](https://github.com/FlowFuse/flowfuse/pull/3025/commits/85bf808b8d71bc4581c7bc75ae75cfbff7ee2d44)